src/pudl/output/ferc1.py

```diff
@@ -1944,12 +1944,25 @@ def forest(self: Self) -> nx.DiGraph:
         remaining_stepparents = set(self.stepparents(forest))
         if remaining_stepparents:
             logger.info(f"{remaining_stepparents=}")
-        # forest = self.set_forest_attributes(
-        #    forest,
-        #    exploded_meta=self.exploded_meta,
-        #    exploded_calcs=self.exploded_calcs,
-        #    tags=self.tags,
-        # )
+
+        # There are a few rare instances where a particular node is specified with more
+        # than one weight (-1 vs. 1) and in those cases, we always want to keep the
+        # weight of -1, since it affects the overall root->leaf calculation outcome.
+        # Maybe this should actually happen in set_forest_attributes?
+        multi_valued_weights = (
+            self.exploded_calcs.groupby(self.calc_cols, dropna=False)["weight"]
+            .transform("nunique")
+            .gt(1)
+        )
+        calcs_to_drop = multi_valued_weights & (self.exploded_calcs.weight == 1)
+        deduplicated_calcs = self.exploded_calcs.drop(calcs_to_drop.index)
```

Comment:
this works fine and i think is a good solution for now, but i think the more holistic way to do this would be to get all of the nodes direct predecessor nodes in order to generate both the parent_cols and calc_cols in order to select the individual records to select on within `exploded_calcs` during the merge in `set_forest_attributes`

Reply:
We should talk about this. I'm not sure I understand what you're saying. I agree this probably belongs over in `set_forest_attributes()` or maybe it its own `deduplicate_calcs()` method that's called by `set_forest_attributes()`